### DISCUSSION TITLE
Regarding transactions

### DISCUSSION THREAD

**Discussion by Biplab-Dutta** [isar/isar#655] [Help]
Is it necessary to use transactions when only dealing with single database operation?
```dart
 Future<void> saveHeader(GithubHeader header) async {
    await isar.githubHeaders.put(header);
  }
```
Here, I am only performing create operation. So, should I be using transaction in such a case?
Will there be performance issue if I didn't use transaction for the case shown above?

**Comment by simc** [comment]
Isar will throw an error if you don't use transactions for write operations.

**Reply by Biplab-Dutta** [reply]
@leisim And what about in the case of a single read operation? Does that have to be under a transaction too?

**Reply by simc** [reply]
No read operations without transactions implicitly create one. Please read the docs, it's all there 🙂

**Comment by parmeetmaster** [comment]
You must try this solution.Just replace **animeListDbInstances** with you instance

    await isar.writeTxn(() async {
      await isar.animeListDbInstances
          .put(AnimeListDbInstance()..setValues(anime));
    });
